core/arch/arm/kernel/virtualization.c

```diff
@@ -322,14 +322,8 @@ uint32_t virt_guest_destroyed(uint16_t guest_id)
 
 bool virt_set_guest(uint16_t guest_id)
 {
-	struct guest_partition *prtn;
-	uint32_t exceptions;
-
-	prtn = get_current_prtn();
-
-	/* This can be true only if we return from IRQ RPC */
-	if (prtn && prtn->id == guest_id)
-		return true;
+	struct guest_partition *prtn = get_current_prtn();
```

Comment:
Just to clarify. Are you sure that foreign IRQ handler calls `virt_unset_guest()`?

I can't find this call and when I wrote this part of the code, this wasn't the case.

Reply:
Yes, that's done in `thread_state_suspend()`. Without this a thread could never migrate to another core.

Reply:
Ah, yes, now I can see it. Thanks.